frontend/src/components/PersistLogin.jsx

```diff
@@ -0,0 +1,45 @@
+import { Outlet } from 'react-router-dom'
+import { useState, useEffect } from 'react'
+import useRefreshToken from '../hooks/useRefreshToken'
+import useAuth from '../hooks/useAuth'
+import { Spinner } from '@chakra-ui/react'
+const PersistLogin = () => {
+  const [isLoading, setIsLoading] = useState(true)
+  const refresh = useRefreshToken()
+  const { auth, persist } = useAuth()
+
+  useEffect(() => {
+    let isMounted = true
```

Comment:
This is to avoid modifying the component when it is yet to be mounted, which may result in an error. Gotten it from the tutorial video that I was following